packages/qwik-city/src/runtime/src/spa-shim.ts

```diff
@@ -6,11 +6,13 @@ import { getPlatform } from '@builder.io/qwik';
 
 import init from './spa-init';
 
+const SPA_SHIM_MINFIED = `const shim=async(t,o,n)=>{if(!window._qcs&&"manual"===history.scrollRestoration){window._qcs=!0;const s=history.state?._qCityScroll;s&&window.scrollTo(s.x,s.y);const i=document.currentScript;if(i){const s=i?.closest("[q\\:container]"),c=new URL(o,new URL(t,document.baseURI));if(isDev){const t=new Function("url","return import(url)");(await t(c.href))[n](s)}else(await import(c.href))[n](s)}}};`;
```

Comment:
You should minify this more, you can pass window, document and history as props in an IIF. Also, the isDev should be evaluated here, not on the client

Reply:
```tsx
(function (w, h, d) {
  const shim = async (base: string, path: string, symbol: string) => {
    if (!(w as ClientSPAWindow)._qcs && h.scrollRestoration === 'manual') {
      // TODO Option to remove this shim especially for MFEs, like loader, for now we only run once.
      (w as ClientSPAWindow)._qcs = true;
  
      const scrollState = (h.state as ScrollHistoryState)?._qCityScroll;
      if (scrollState) {
        w.scrollTo(scrollState.x, scrollState.y);
      }
  
      const script = d.currentScript as HTMLScriptElement;
      if (script) {
        // Inside shadow DOM, we can't get a hold of a container. So we can't
        // load the SPA shim.
        const container = script!.closest('[q\\:container]')!;
        const url = new URL(path, new URL(base, d.baseURI));
  
        if (isDev) {
          // Bypass dev import hijack. (not going to work here)
          // eslint-disable-next-line no-new-func
          const imp = new Function('url', 'return import(url)');
          (await imp(url.href))[symbol](container);
        } else {
          (await import(url.href))[symbol](container);
        }
      }
    }
  };
})(window, history, document);
```

Does this look correct before I minify it? 
![image](https://github.com/user-attachments/assets/18ef2739-45da-461c-a017-68b18d31f2c0)

Reply:
I think so 😅

Reply:
oh wait I didn't stringify it

```tsx
(function (w, h, d) {
  const shim = async (base: string, path: string, symbol: string) => {
    if (!(w as ClientSPAWindow)._qcs && h.scrollRestoration === 'manual') {
      // TODO Option to remove this shim especially for MFEs, like loader, for now we only run once.
      (w as ClientSPAWindow)._qcs = true;

      const scrollState = (h.state as ScrollHistoryState)?._qCityScroll;
      if (scrollState) {
        w.scrollTo(scrollState.x, scrollState.y);
      }

      const script = d.currentScript as HTMLScriptElement;
      if (script) {
        // Inside shadow DOM, we can't get a hold of a container. So we can't
        // load the SPA shim.
        const container = script!.closest('[q\\:container]')!;
        const url = new URL(path, new URL(base, d.baseURI));

        if (isDev) {
          // Bypass dev import hijack. (not going to work here)
          // eslint-disable-next-line no-new-func
          const imp = new Function('url', 'return import(url)');
          (await imp(url.href))[symbol](container);
        } else {
          (await import(url.href))[symbol](container);
        }
      }
    }
  };
  shim.toString();
})(window, history, document);
```